subsys/bootloader/bl_storage/bl_storage.c

```diff
@@ -92,6 +94,17 @@ int public_key_data_read(u32_t key_idx, u8_t *p_buf, size_t buf_size)
 		return -EFAULT;
 	}
 
+	for (u32_t n = 0; n < num_public_keys_read(); n++) {
+		const u8_t *p_key_n = p_bl_storage_data->key_data[n].hash;
```

Comment:
Would it improve readability if this is a `const u16_t *p_key_halfword_n` so we don't need the cast when giving the argument to `read_halfword`?